app/connectors/GoodsRecordConnector.scala

```diff
@@ -111,6 +111,20 @@ class GoodsRecordConnector @Inject() (config: Configuration, httpClient: HttpCli
       .execute[HttpResponse]
       .map(_ => Done)
 
+  def updateSupplementaryUnitForGoodsRecord(
+    eori: String,
+    recordId: String,
+    supplementaryRequest: SupplementaryRequest
+  )(implicit
+    hc: HeaderCarrier
+  ): Future[Done] =
+    httpClient
+      .patch(goodsRecordUrl(eori, recordId))
+      .setHeader(clientIdHeader)
+      .withBody(Json.toJson(UpdateRecordRequest.mapFromSupplementary(supplementaryRequest)))
+      .execute[HttpResponse]
```

Comment:
Just a thought that if @pauloncast  beats you to the merge you might need to change this to use executeAndContinue

Reply:
Poor @pauloncast , sorry about it :p